### DISCUSSION TITLE
Pico W : WebREPL : Ctrl-C not working

### DISCUSSION THREAD

**Discussion by villeneuve** [micropython/micropython#11532] [RP2040 / Pico]
Hi!
With latest 1.20 (also tested with today nightly build) pressing Ctrl-C in WebREPL on a pico W that is running a program has no effect.
Ctrl-C works in normal REPL but not in WebREPL
On ESP8266 and ESP32 (latest 1.20 as well) it works as expected in WebREPL
And also (on the pico W) webrepl_setup exits on error because there is no boot.py file.
We can just create an empty one and it works but this isn't obvious for beginers (boot.py exists on ESP* ports)

**Comment by villeneuve** [comment]
I wanted to try micropython.kbd_intr(chr) to check if it can change something but I found it doesn't work as per the [documentation](https://docs.micropython.org/en/latest/library/micropython.html#micropython.kbd_intr)
micropython.kbd_intr(-1) doesn't disable Ctrl-C
micropython.kbd_intr(3) doesn't enable it (in WebREPL)

MicroPython v1.20.0-68-g3229791b6 on 2023-05-11; Raspberry Pi Pico W with RP2040
Type "help()" for more information.
>>> import micropython
>>> micropython.kbd_intr(-1)
>>> import blink
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "blink.py", line 6, in <module>
KeyboardInterrupt: 
>>>

**Reply by robert-hh** [reply]
Each command entered in REPL enables keyboard interrupt again. In your case it is the import statement. You have to disable keyboard interrupt in your blink.py script.

**Reply by villeneuve** [reply]
Yes Robert you are right, I tested it and it works.
Blame me because I didn't correctly read the doc! 
The doc says: "during script execution"

But it fails to (re)enable Ctrl-C in WebREPL.

WebREPL permits user's application modifications/updates OTA.
It's very convenient for devices without possible physical access.
However if main.py cannot be interrupted it's lost.
There are workarounds, for example the device can respond to mqtt commands to exit main.py
But anyway Ctrl-C is convenient, so why is missing in rp2 port? 
Is it a bug or a feature that will be available in a next version?

Thanks

**Reply by robert-hh** [reply]
I see that Ctrl-C interrupt is missing in webREPL, and it's surely a needed feature.

**Reply by robert-hh** [reply]
There is PR #9297 which implements it, and it works for Ctrl-C, but has other problems. But that PR is not merged.

**Reply by villeneuve** [reply]
Thank you Robert, for pointing this PR, I didn't see it.
I just read it, as the author and you say there are other problems.
So let's wait until the problems are solved and the PR merged.
Then hopefully Ctrl-C will work as on other platforms.

**Reply by felixdoerre** [reply]
So, despite lacking feedback from the maintainers I revisited the PR and in my view I can be merged now. But without any maintainer feedback/reaction there is nothing I can do.
